frontend/packages/console-dynamic-plugin-sdk/src/api/k8s-types.ts

```diff
@@ -50,3 +57,19 @@ export type K8sList = <R extends K8sResourceCommon>(options: {
 }) => Promise<R[]>;
 
 export type GetK8sResourcePath = (model: K8sModel, options: Options) => string;
+
+export type GetReference = (K8sGroupVersionKind: K8sGroupVersionKind) => K8sResourceKindReference;
```

Comment:
this needs to be removed too as anything new (function/type/whatever) that is somehow using `K8sResourceKindReference`

Reply:
Sorry, I thought we need `getReference`  as we still need to form `K8sResourceKindReference` internally for the hooks if the user passes `K8sGroupVersionKind` ? or do you mean to remove type?

Reply:
having these utility functions/types is ok as long as they are used only internally and not exposed. So these types /utilities should not be accessible via `lib-core.ts` (which they currently are).

We could move them to `src/utils` probably

Reply:
ah, all types are exposed by default I see. Have made it an implicit type in function which is in utils and removed it.